internal/integration/clam_prose_test.go

```diff
@@ -199,8 +199,8 @@ func clamMultiByteTruncLogs(mt *mtest.T) []truncValidator {
 	// Insert started.
 	validators[0] = newTruncValidator(mt, cmd, func(cmd string) error {
 
-		// Remove the suffix from the command string.
-		cmd = cmd[:len(cmd)-len(logger.TruncationSuffix)]
+		// Remove the suffix and "}" bracket from the command string.
+		cmd = cmd[:len(cmd)-len(logger.TruncationSuffix)-1]
```

Comment:
Document.StringN() should not apply a bracket if it had to truncate data. This is caused by this block:

```go
if buf.Len() <= n {
	buf.WriteByte('}')
}
```

AFAIK truncating will always ensure this is always true. The check to truncate, `buf.Len()+len(str) > n`, will only be true if truncation will occur. Since this implies that the length of the string is always greater than the truncation width: $\text{size} = L_s > n - B = \text{width}$. So the algorithm can be adjusted like so:

```go
truncated := false
if n > 0 {
	for length > 1 {
		// ...
		str := elem.StringN(n)
		if buf.Len()+len(str) > n {
			str := truncate(str, uint(n-buf.Len()))
			buf.WriteString(str)

			truncated = true
			break
		}
	
		// ... 
	}
}

if !truncated {
	buf.WriteByte('}')
}
```